src/interfaces/IERC20.sol

```diff
@@ -36,6 +36,21 @@ interface IERC20 {
      */
     function allowance(address owner, address spender) external view returns (uint256);
 
+    /**
+     * @dev Returns the name of the token.
+     */
+    function name() external view returns (string memory);
+
+    /**
+     * @dev Returns the symbol of the token.
+     */
+    function symbol() external view returns (string memory);
+
+    /**
+     * @dev Returns the amount of decimal points for displaying a token amount.
+     */
+    function decimals() external view returns (uint8);
```

Comment:
So, it seems that `IERC20` doesn't require these methods. But, there is `IERC20Metadata` that adds these extensions. Let's [mimic](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#IERC20Metadata) having two separate interfaces that our ERC20 contracts implement? This way we don't assume that these [optional methods](https://eips.ethereum.org/EIPS/eip-20#methods) are required for external tokens.

Reply:
To be fair, these are all optional in the ERC20 standard so none of these should be part of `IERC20`.

Reply:
Got it. I will remove the extras then